### PR TITLE
Add rest endpoints for managing thumbnail files.

### PR DESCRIPTION
**Pull request by manthey**
Add the ability to delete thumbnail files (this facilitates testing and frees disk space).

Add an endpoint that starts a local job that creates thumbnail files.

This resolves issue #83, though we could want a setting to run a cron-like job at some point.

### CODE REVIEW

**Comment by manthey** [comment]
This is ready for review.

File: plugin_tests/common.py [hunk 1]
```diff
@@ -224,6 +235,11 @@ def _postTileViaHttp(self, itemId, fileId):
         self.assertTrue('Item already has' in req.json()['message'] or
                         'Item is scheduled' in req.json()['message'])
 
+        if jobAction == 'delete':
+            jobModel = self.model('job', 'jobs')
+            jobModel.remove(jobModel.find(
+                {}, sort=[('_id', SortDir.DESCENDING)])[0])
```
Comment (zachmullen) [inline plugin_tests/common.py]: This is an odd construct, what's happening here?

File: server/models/image_item.py [hunk 2]
```diff
@@ -256,6 +258,8 @@ def getThumbnail(self, item, width=None, height=None, **kwargs):
         """
         # check if a thumbnail file exists with a particular key
         keydict = dict(kwargs, width=width, height=height)
+        for key in [key for key in keydict if keydict[key] is None]:
```
Comment (zachmullen) [inline server/models/image_item.py]: A dict comprehension would probably be more readable here.

    keydict = {k: v for k, v in six.viewitems(keydict) if v is not None}

File: plugin_tests/common.py (hunk 1 above)
Comment (manthey) [inline plugin_tests/common.py]: This deletes the most recently created job.  Mongo's _ids increase monotonically, so the sort returns the highest (most recent) first.

**Pull request merged**